### DISCUSSION TITLE
Jagged Icons

### DISCUSSION THREAD

**Discussion by jjanuschv** [elrumo/macOS_Big_Sur_icons_replacements#1119] [Help with the website]
I'm noticing a not-insignificant number of icons I'm applying are really jagged when applied. It's not all of them, but a good number. You can see it in the Info window:

![image](https://user-images.githubusercontent.com/76971259/111687622-b3e11580-87e7-11eb-98d2-2dfdc2a6cfdd.png)

The changes persist into the dock and when CMD+Tabbing. This is the icon in question:

![image](https://user-images.githubusercontent.com/76971259/111687821-f73b8400-87e7-11eb-931b-3a1633c952c5.png)
https://onionicons.com/parse/files/macOSicons/4cf2431653fdb8793b8d570fa1304fba_1606693642841_Slack.icns

Is anyone else experiencing this? Or is it due to some setting I have or something?

**Comment by rodgerwerner** [comment]
This icon for sure is jagged when viewed smaller. When viewed in Preview the sizes around 256px and below get quite jagged or aliased. This, more than likely, has something to do with how the icon is made on the server. When I take the 1024px size out of this icon and use it as reference to make an icon using iconutil, the icon at smaller sizes does not exhibit this behaviour.

**Reply by jjanuschv** [reply]
It seems you were right. I regenerated the bad icons using iconutil and they look a lot better (although they are definitely blurry at lower resolutions; the downside of downscaling).

If anyone is reading this, I created a simple bash script based on [this](https://stackoverflow.com/a/20703594/1515620) StackOverflow question to regenerate these:

```sh
#!/bin/sh

PNG=$1
ICNS=$2

mkdir TmpIcon-1234567890.iconset
sips -z 16 16     $PNG --out TmpIcon-1234567890.iconset/icon_16x16.png
sips -z 32 32     $PNG --out TmpIcon-1234567890.iconset/icon_16x16@2x.png
sips -z 32 32     $PNG --out TmpIcon-1234567890.iconset/icon_32x32.png
sips -z 64 64     $PNG --out TmpIcon-1234567890.iconset/icon_32x32@2x.png
sips -z 128 128   $PNG --out TmpIcon-1234567890.iconset/icon_128x128.png
sips -z 256 256   $PNG --out TmpIcon-1234567890.iconset/icon_128x128@2x.png
sips -z 256 256   $PNG --out TmpIcon-1234567890.iconset/icon_256x256.png
sips -z 512 512   $PNG --out TmpIcon-1234567890.iconset/icon_256x256@2x.png
sips -z 512 512   $PNG --out TmpIcon-1234567890.iconset/icon_512x512.png
cp $PNG TmpIcon-1234567890.iconset/icon_512x512@2x.png
iconutil -c icns TmpIcon-1234567890.iconset --output $2
rm -R TmpIcon-1234567890.iconset
```

To run:

1. Open icon in Preview
2. Select highest resolution icon and then right click -> Export and export it as a PNG
3. Run `png2icns icon.png icon.icns` where `png2icns` is the name of the script you saved, `icon.png` is the PNG you exported, and `icon.icns` is the ICNS you wish to generate (must end in `.icns`)

**Reply by eproxus** [reply]
Also note that you can open the high-resolution PNG in Preview, select *Edit → Copy* and click the icon in the *Get Info* window of the app and paste with ⌘+V. That will use macOS internal scaling which looks really good. No need to use an ICNS file.

@elrumo Can we get PNG download links on the website as well?

**Comment by colineberly** [comment]
I've fixed a bunch of these. It looks like the generated icons are getting pixelated @1x but are fine @2x. There is a really simple way to fix them if using a program like Icon Slate for instance. In Icon Slate you can pull up the icon, duplicate it, right-click "Clear all sizes" go to @2x size of the original and copy or drag it in to the large box. This will repopulate all sizes and you can export the .icns file then and there. Obviously, this works for local icons. Not sure what the fix is for the server side...

**Reply by elrumo** [reply]
The solution on the server-side is to re-export the icons, new icons should be fine because I changed the resizing algorithm. Right now I'm working on getting the site working again after someone hacked it and ransomed the database...

**Reply by eproxus** [reply]
@elrumo I think the server resizing is still not perfect (unless you fixed something in the last few days again). I get better results from macOS' internal scaling when pasting the 1024 px PNG icon onto an application than when using a newly uploaded ICNS file.
